src/__tests__/metatests.js

```diff
@@ -73,3 +74,109 @@ describe('jsdom-global', () => {
     }
   });
 });
+
+describe('lolex', () => {
+  let lolex: Lolex;
+
+  beforeAll(() => {
+    lolex = new Lolex();
+  });
+
+  afterAll(() => {
+    lolex.dispose();
+  });
+
+  afterEach(() => {
+    // clear any unset timers
+    lolex.clearAllTimers();
+  });
+
+  test('Date.now() is mocked by Lolex', () => {
+    const start = Date.now();
+    lolex.advanceTimersByTime(5.5e9); // ~63.66 days
+    const end = Date.now();
+
+    const apparentDuration = end - start;
+
+    expect(apparentDuration).toBeGreaterThan(5e9);
+    expect(apparentDuration).toBeLessThan(6e9);
+  });
+
+  test('setInterval is triggered by Lolex', () => {
+    let count = 0;
+    setInterval(() => {
+      ++count;
+    }, 1e9);
+
+    lolex.advanceTimersByTime(5.5e9);
+
+    expect(count).toBe(5);
+  });
+
+  test('setTimeout is triggered by Lolex', () => {
+    let flag = false;
+    setTimeout(() => {
+      flag = true;
+    }, 3e9);
+    setTimeout(() => {
+      flag = true;
+    }, 6e9);
+
+    lolex.advanceTimersByTime(5.5e9);
+
+    expect(flag).toBe(true);
+  });
```

Comment:
Perhaps make the second timeout set the flag to `false`? That'd make the test slightly more informative; otherwise, it seems like it'd be more informative if the second timeout were deleted.

Reply:
> That'd make the test slightly more informative

This is incorrect: it makes the test *much* more informative.

It reveals a bug, you see. :disappointed:

Reply:
**Addendum:** The bug involves Lolex quietly converting timeout values greater than 2^31 - 1 into very short timeouts, to match Node's behavior. (Allegedly the cap here is part of the [HTML spec](https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#timers), but I can't find language implying it. Modern Chrome and Firefox use the 32 least significant bits rather than rolling to 0.)

Reply:
Well then! Glad that was informative. :slightly_smiling_face: